linkerd/protocol/h2/src/main/scala/com/twitter/finagle/buoyant/h2/LinkerdHeaders.scala

```diff
@@ -435,13 +435,31 @@ object LinkerdHeaders {
   }
 
   class ClearMiscServerFilter extends SimpleFilter[Request, Response] {
-    def apply(req: Request, service: Service[Request, Response]) = {
-      for ((k, _) <- req.headers.toSeq) {
+
+    private[this] def clearLinkerdHeaders(headers: Headers) = {
+      for ((k, _) <- headers.toSeq) {
         if (k.toLowerCase.startsWith(LinkerdHeaders.Prefix)) {
-          req.headers.remove(k)
+          headers.remove(k)
+        }
+      }
+    }
+
+    def apply(req: Request, service: Service[Request, Response]) = {
+      clearLinkerdHeaders(req.headers)
+      service(req).map { resp =>
+        val rsp = resp.dup()
+        if (rsp.headers.contains(Err.Key)) {
+          Stream.readToEnd(rsp.stream)
```

Comment:
maybe write this as `val _ = Stream.readToEnd(rsp.stream)` to explicitly signal that we're doing a fire-and-forget future here.  Or add a comment.